### ISSUE TITLE
New Command: Get-DbaTraceFlagIndex

### ISSUE THREAD

**Issue by wsmelton** [sqlcollaborative/dbatools#2300]
## Feature Request
Provide the ability to have a central list of all known trace flags for SQL Server. Build it out similar to the `Find-DbaCommand` where we have an `index.json` file that contains pertinent information...just as a starting point:

- TraceFlag number
- Description/Synopsis of use
- SQL Server version applied to
- Link (KB, or default docs page)

### Problem to solve

Having to look up a TF for a particular need or fix.

**Comment by michalporeba** [comment]
What is the latest thinking on this? 
I normally go to https://docs.microsoft.com/en-us/sql/t-sql/database-console-commands/dbcc-traceon-trace-flags-transact-sql if I need information on trace flags. 

One option would be to parse that page when building the `traceflags.json` file, the other would be to simply create the file and a method that reads it. The trace flags are not added that frequently, and the existing ones don't change a lot.

**Comment by gbargsley** [comment]
I had played with this and was able to scrape the web page to get the trace flag and descriptions info. However, I got stuck on how to load to json file and search the file with the function. Never had a chance to revisit.

**Comment by wsmelton** [comment]
It is the same consideration as the build reference and the data we keep on that, or the wait type information we include in commands (links to SQLSkills site). However, it is more or less to also include undocumented trace flags that are not included in the docs but other resources speak/write on them and the usage.

Other references:
- https://www.brentozar.com/blitz/trace-flags-enabled-globally/
- http://www.sqlservercentral.com/articles/trace-flag/152989/
- http://www.sqlservice.se/updated-microsoft-sql-server-trace-flag-list/

**Comment by wsmelton** [comment]
Excellent info in this one: https://dba.stackexchange.com/q/141092/507

Check out the links that Paul White shares, that PDF download is stinking awesome.

**Comment by potatoqualitee** [comment]
This is updated on an ongoing basis, as recently as March 24th. But it looks hard to parse.

https://github.com/ktaranov/sqlserver-kit/blob/master/SQL%20Server%20Trace%20Flag.md

**Comment by potatoqualitee** [comment]
Carefully considering new commands as each bit of code slows down the import and gets lost in the mix. This one does not make it bc there's a markdown available online 🙏🏼
